src/CodeGen_C.cpp

```diff
@@ -2845,9 +2876,7 @@ void CodeGen_C::visit(const Shuffle *op) {
     if (op->type.is_scalar()) {
         rhs << src << "[" << op->indices[0] << "]";
     } else {
-        string indices_name = unique_name('_');
-        stream << get_indent() << "const int32_t " << indices_name << "[" << op->indices.size() << "] = { " << with_commas(op->indices) << " };\n";
-        rhs << print_type(op->type) << "::shuffle(" << src << ", " << indices_name << ")";
+        rhs << print_type(op->type) << "::shuffle<" << with_commas(op->indices) << ">(" << src << ")";
```

Comment:
A thought from my last pass through this: should we just generate code like:

    Vec result = { operand[indices_0], operand[indices_1], ... };

? Or at least in the case we lack builtin_shuffle?

Reply:
Is that likely to be a win in terms of the machine code we generate?

Reply:
I think it's a win in that we generate less code for the compiler to parse and optimize. The final machine code probably wouldn't be very different. 

I would expect the code I proposed to be more likely to get optimized as a shuffle, but I would think they both do.

Reply:
Yeah, just thinking about it more though... realistically, I don't think we should ever even see that case. If we have native vector support, we're gonna have __builtin_shuffle or __builtin_shufflevector (for gcc vs clang). Putting `#error` there is probably simpler.

(MSvC supports none of this, of course...)